cmd/htcondor-api/main.go

```diff
@@ -126,6 +126,37 @@ func runNormalMode() error {
 		logger.Info(logging.DestinationCollector, "Created collector", "host", collectorHost)
 	}
 
+	// MCP configuration
+	enableMCP := false
+	oauth2DBPath := ""
+	oauth2Issuer := ""
+
+	// Check if MCP should be enabled from config
+	if mcpEnable, ok := cfg.Get("HTTP_API_ENABLE_MCP"); ok && mcpEnable == "true" {
+		enableMCP = true
+		log.Println("MCP endpoints enabled via configuration")
+	}
+
+	// Get OAuth2 DB path from config, default to /var/lib/condor/oauth2.db
+	if enableMCP {
+		if dbPath, ok := cfg.Get("HTTP_API_OAUTH2_DB_PATH"); ok && dbPath != "" {
+			oauth2DBPath = dbPath
+		} else if localDir, ok := cfg.Get("LOCAL_DIR"); ok && localDir != "" {
+			oauth2DBPath = filepath.Join(localDir, "oauth2.db")
+		} else {
+			oauth2DBPath = "/var/lib/condor/oauth2.db"
+		}
+		log.Printf("OAuth2 database path: %s", oauth2DBPath)
+
+		// Get OAuth2 issuer from config or construct from listen address
+		if issuer, ok := cfg.Get("HTTP_API_OAUTH2_ISSUER"); ok && issuer != "" {
+			oauth2Issuer = issuer
+		} else {
+			oauth2Issuer = "http://" + listenAddrFromConfig
```

Comment:
This likely results in the IP address being used.  Default to `FULL_HOSTNAME` instead of `listenAddrFromConfig` and always default to `https://`.